DRODLib/GameConstants.cpp

```diff
@@ -60,32 +60,72 @@ const WCHAR wszVersionReleaseNumber[] = {
 //Schema of player input commands
 namespace InputCommands
 {
-	const char *COMMANDNAME_ARRAY[DCMD_Count] = {
-		"MoveNorthwest",
-		"MoveNorth",
-		"MoveNortheast",
-		"MoveWest",
-		"Wait",
-		"MoveEast",
-		"MoveSouthwest",
-		"MoveSouth",
-		"MoveSoutheast",
-		"SwingClockwise",
-		"SwingCounterclockwise",
-		"Restart",
-		"Undo",
-		"Battle",
-		"UseCommand",
-		"CloneSwitch"
-	};
-
-	const UINT COMMAND_MIDS[DCMD_Count] = {
-		MID_MoveNorthwest, MID_MoveNorth, MID_MoveNortheast,
-		MID_MoveWest, MID_Wait, MID_MoveEast,
-		MID_MoveSouthwest, MID_MoveSouth, MID_MoveSoutheast,
-		MID_SwingClockwise, MID_SwingCounterclockwise,
-		MID_RestartRoom, MID_UndoMove, MID_BattleKey, MID_UseCommandKey, MID_CloneSwitch
-	};
+	const std::unordered_map<DCMD, KeyDefinition*> BuildKeyDefinitions()
+	{
+		std::unordered_map<DCMD, KeyDefinition*> keyDefinitions;
+		
+		keyDefinitions[DCMD_NW] = new KeyDefinition(CMD_NW, "MoveNorthwest", MID_MoveNorthwest, SDLK_KP_7, SDLK_7);
```

Comment:
Having everything for each definition laid out all in one place is much nicer than what we had before!

Reply:
Also it's less error prone :).